xbmc/music/windows/GUIWindowMusicBase.cpp

```diff
@@ -690,12 +692,20 @@ bool CGUIWindowMusicBase::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
       Update(item->GetPath());
       int bookmark;
       m_musicdatabase.GetResumeBookmarkForAudioBook(item->GetPath(), bookmark);
-      int i=0;
-      while (i < m_vecItems->Size() && bookmark > m_vecItems->Get(i)->m_lEndOffset)
-        ++i;
-      CFileItem resItem(*m_vecItems->Get(i));
-      resItem.SetProperty("StartPercent", ((double)bookmark-resItem.m_lStartOffset)/(resItem.m_lEndOffset-resItem.m_lStartOffset)*100);
-      g_application.PlayFile(resItem, "", false);
+
+      auto itemIt = std::find_if(
+        m_vecItems->begin(),
+        m_vecItems->end(),
+        [&](const CFileItemPtr& item) { return bookmark <= item->m_lEndOffset; }
+      );
+      if(itemIt != m_vecItems->end())
```

Comment:
```suggestion
      if (itemIt != m_vecItems->end())
```

sorry, slipped through - everything else is good to go :+1: